components/layout_thread/lib.rs

```diff
@@ -673,7 +682,12 @@ impl LayoutThread {
                 self.paint_time_metrics.maybe_set_metric(epoch, paint_time);
                 true
             },
-            Request::FromScript(msg) => self.handle_request_helper(msg, possibly_locked_rw_data),
+            Request::FromScript(msg) => {
+                self.busy.store(true, Ordering::Relaxed);
```

Comment:
Rather than only marking when we process script messages, we should be setting this to false before the select! and setting it to true immediately afterwards.

Reply:
Fixed this in 8b88526.